dot/core/service.go

```diff
@@ -398,6 +400,59 @@ func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
 	return nil
 }
 
+// ProcessBlockRequestMessage processes a block request message, returning a block response message
+func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
+	blockRequest := msg.(*network.BlockRequestMessage)
+
+	latestBlockNum := s.blockState.LatestHeader().Number
+
+	msgStartingBlockPrefix := blockRequest.StartingBlock[0]
+	msgStartingBlockBody := blockRequest.StartingBlock[1:]
+
+	var msgStartingBlockNum *big.Int
+	var msgBlockData *types.BlockData
+	var err error
+
+	if msgStartingBlockPrefix == 1 {
+		msgStartingBlockNum = big.NewInt(0).SetBytes(msgStartingBlockBody)
+		// GetBlockData by block number ?
+	} else if msgStartingBlockPrefix == 1 {
+
+		msgBlockData, err = s.blockState.GetBlockData(common.NewHash(msgStartingBlockBody))
+		if err != nil {
+			log.Debug("(*Service).ProcessBlockRequestMessage failed to GetBlockData with StartingBlock", "StartingBlock", common.NewHash(blockRequest.StartingBlock[1:]))
+			return err
+		}
+	} else {
+		err = fmt.Errorf("(*Service).ProcessBlockRequestMessage blockRequest.StartingBlock not recognized")
+		log.Trace(err.Error())
+		return err
+	}
+
+	// for now we only care about body requests
+	if blockRequest.RequestedData == 1 && latestBlockNum.Cmp(msgStartingBlockNum) >= 0 {
```

Comment:
this should use a bitmap comparison to check the requested data, for example, 2, 6, 10, etc. are all requests for the block body

you should do `blockRequest.RequestedData & 1` to check for header request, blockRequest.RequestedData & 2` for body request, etc